packages/search-in-workspace/src/browser/quick-search-in-workspace.ts

```diff
@@ -0,0 +1,230 @@
+                        },
+
+                        onDone: (searchId: number, error?: string) => {
+                                if (searchId !== this.currentSearchId) {
+                                        this.logger.debug('Search ' + this.currentSearchId + ' has completed, but it\'s not the current search.');
+                                        return;
+                                }
+                                this.logger.debug('Search ' + this.currentSearchId + ' has completed and is the current search.');
+                                this.currentSearchId = -1;
+
+                                if (error) {
+                                        this.showFakeResult(error, acceptor);
+                                } else if (items.length !== 0) {
+                                        items.sort((a, b) => SearchInWorkspaceResultCompare(a.getResult(), b.getResult()));
+                                        acceptor(items);
+                                } else {
+                                        this.showFakeResult('No matches :(', acceptor);
+                                }
+
+                        },
+                }, opts).then(searchId => {
+                        this.currentSearchId = searchId;
+                });
+        }
+
+        showFakeResult(label: string, acceptor: (items: QuickOpenItem[]) => void) {
+                acceptor([
+                        new QuickOpenItem({
+                                label: label,
+                        }),
+                ]);
+        }
+
+        /* If we have an ongoing search, cancel it.  */
+        cancelCurrentSeach() {
+                if (this.currentSearchId >= 0) {
+                        this.logger.debug('Cancelling search ' + this.currentSearchId);
+                        this.searchInWorkspaceService.cancel(this.currentSearchId);
+                        this.currentSearchId = -1;
+                }
+        }
+
+        /* Open the quick search in workspace popup.  */
+        open() {
+                const opts: QuickOpenOptions = {
+                        onClose: cancelled => this.cancelCurrentSeach(),
+                        placeholder: 'Search in workspace by regular expression...',
+                };
+                this.quickOpenService.open(this, opts);
+        }
+}
+
+class QuickSearchInWorkspaceResultItem extends QuickOpenItem {
+
+        private result: SearchInWorkspaceResult;
+        private openerService: OpenerService;
+
+        constructor(result: SearchInWorkspaceResult, openerService: OpenerService, labelProvider: LabelProvider) {
+                const hl: Highlight = {
+                        start: result.character - 1,
+                        end: result.character + result.length - 1,
+                };
+
+                /* Show the path relative to the workspace.  */
+                const file = labelProvider.getLongName(new URI('file://' + result.file));
```

Comment:
we should show the simple file name. The path should be shown after it.
i.e.
**my-functions.ts:30** _src/mypackage_

starting with the path is too much noise before the important segment.
Also, I think the column is not helpful and should be skipped.

Reply:
Ok, I'll work on that.  Agreed that the column is not really useful.